src/app/games/connect-six/tests/ConnectSixMinimax.spec.ts

```diff
@@ -0,0 +1,169 @@
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, O, O, O, O, O, _, _, _, _, _, _, _, _, _, _, _, _],
+        ];
+        const state: ConnectSixState = new ConnectSixState(board, 3);
+        const node: ConnectSixNode = new ConnectSixNode(state);
+
+        // When asking what is the best move
+        const bestMove: ConnectSixMove = node.findBestMove(2, minimax);
+
+        // Then the minimax level two should block
+        expect(bestMove).toEqual(ConnectSixDrops.from(new Coord(1, 18), new Coord(7, 18)).get());
+    });
+    it('should count all possible moves including only neighbooring-coord', () => {
+        // Given a board with 60 possibles combinaisons of two coords
+        // With the first being neighbors of the original set
+        // and the second able to also be a neighboor of the first coord
+        const board: Table<PlayerOrNone> = [
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _, _],
+            [_, _, O, O, O, O, O, _, _, _, _, _, _, _, _, _, _, _, _],
+        ];
+        const state: ConnectSixState = new ConnectSixState(board, 3);
+        const node: ConnectSixNode = new ConnectSixNode(state);
+
+        // When counting the number of choices
+        const moves: ConnectSixMove[] = minimax.getListMoves(node);
+
+        // Then the answer should be 60
+        expect(moves.length).toBe(60);
+    });
+    it('should block double-open four at level two', () => {
+        // Given a minimax at level two
+        // And an board where current opponent could win if nothing is done (..XXXX..)
```

Comment:
same comment

Reply:
a board where opponent must be blocked [in order for current player to avoid loosing] ?

Reply:
yep

Reply:
euh, j'ai updaté comme dans l'autre commentaire